### ISSUE TITLE
Maven -> Update Projects vs checked in files

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext#2227]
Maven -> Update Projects changes checked in files `.classpath`, `.project`. these might also leave projects uncompilable e.g. generatorxxx folders in xtend.core vanish

we should check what we can do about this

**Comment by LorenzoBettini** [comment]
Yes, I know. In fact, in oomph setup I disabled the corresponding options. As far as I know, there are a few open bugs in m2e. For the moment, there's not much we can do about that.

 Concerning the generator folders in xtend core maybe we can configure build helper maven plugin appropriately to add those folders as source folders..

**Comment by HannesWell** [comment]
On my Windows computer I also notice that some extend files are generated with different line separators in strings compared to how they are committed. This leads to about 8 changed files in my Git-Staging when I run a full workspace build.
I wonder if those xtend-generated files can be ignored by git, if they are generated anyways.


> Yes, I know. In fact, in oomph setup I disabled the corresponding options. As far as I know, there are a few open bugs in m2e. For the moment, there's not much we can do about that.

Can you point me to the exact bugs?

**Comment by cdietrich** [comment]
Regarding Xtend

https://github.com/eclipse/xtext-xtend/issues/872
https://github.com/xtext/xtext-monorepo/issues/12

**Comment by LorenzoBettini** [comment]
> Can you point me to the exact bugs?

IIRC,
https://github.com/eclipse-m2e/m2e-core/issues/1200
https://github.com/eclipse-m2e/m2e-core/issues/1239
https://github.com/eclipse-m2e/m2e-core/issues/1278

**Comment by LorenzoBettini** [comment]
Concerning generatorxxx folders in xtend.core, I guess that is due to the `build.properties`:

```
src.excludes = generator/,\
               generator-xtend-gen/
```

and I guess we want it that way?

**Comment by cdietrich** [comment]
we want to have this a source folders but not package. also building with maven is not a must

**Comment by LorenzoBettini** [comment]
I don't understand the last part.

**Comment by cdietrich** [comment]
we dont run the workflow from maven. so there is no strict need to compile with maven. i also dont know if it would cause of the jflex thing

**Comment by LorenzoBettini** [comment]
but I don't understand what you propose...

**Comment by cdietrich** [comment]
i propose nothing.

**Comment by szarnekow** [comment]
Is it possible in maven to have something next to `src/main` and `src/test`? What we effectively need is `src/generator`, so a source folder that is neither test nor production.

**Comment by mickaelistria** [comment]
The issue comes from m2e, as linked by @LorenzoBettini earlier. PRs to fix that in m2e would be highly welcome!

**Comment by mickaelistria** [comment]
FWIW, I investigated it in m2e and the .project changes proposed by m2e do happen from time to time, but are stable: as long as the Maven Nature is present and Maven Builder is the last one defined in the .project, m2e shouldn't modify the .project. So maybe you can just commit those .project is m2e has "fixed" them (it can happen eg if you enable another builder, it usually becomes the last on the list and then m2e will want to modify your .project as long as it's not the last one running).
For the .classpath, I would recommend that one usually does *not* check them into SCM unless they have specific manually maintained configuration in those. If the pom.xml are configured in a standard enough way, the .classpath will always be correct and regenerated on project update. So it's often better to just not check it in and .gitignore it.
